lib/fs.js

```diff
@@ -1867,6 +1867,49 @@ WriteStream.prototype._write = function(data, encoding, cb) {
 };
 
 
+function writev(fd, chunks, position, callback) {
+  function wrapper(err, written) {
+    // Retain a reference to chunks so that they can't be GC'ed too soon.
```

Comment:
I'm following the other examples in fs.js that have this pattern. I did not invent this to be honest.
My guess is that not having the chunks (that contain buffers) in this closure would GC and free the buffers before libuv has a chance to access and use them.